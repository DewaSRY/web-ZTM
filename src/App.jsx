import { useState, useEffect } from "react";
import Card from "./Component/Card-components";
import SerchBox from "./Component/Serch-Box-components";
import "./App.css";

const App = () => {
  const [sherchRobot, setSerchRobots] = useState("");
  const [robots, setRobots] = useState([]);
  const [filterRobots, ssetFilterRobots] = useState(robots);
  useEffect(() => {
    fetch("https://jsonplaceholder.typicode.com/users")
      .then((response) => response.json())
      .then((users) => setRobots(users));
  }, []);
  useEffect(() => {
    const newSrechRobot = robots.filter((robot) =>
      robot.name.toLocaleLowerCase().includes(sherchRobot)
    );
    ssetFilterRobots(newSrechRobot);
  }, [robots, sherchRobot]);

  const sechHeandler = (event) => {
    const sherch = event.target.value.toLowerCase();
    setSerchRobots(sherch);
  };
  const RobotsDisplay = () => {
    return (
      <div className="tc">
        <h1>ROBOT FRIRENDS</h1>
        <SerchBox onSherch={sechHeandler} />
        <div>
          {filterRobots.map((robots) => (
            <Card key={robots.id} robots={robots} />
          ))}
        </div>
      </div>
    );
  };

  return <>{robots ? <RobotsDisplay /> : <p>Loading...</p>}</>;
};

export default App;
