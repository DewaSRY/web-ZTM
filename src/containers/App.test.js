import { shallow } from "enzyme";
import App from "./App";

it("test app container", () => {
  const mockStore = {
    robots: [],
    searchfield: "",
  };
  const test = shallow(<App store={mockStore} />);
  expect(test).toMatchSnapshot();
});
