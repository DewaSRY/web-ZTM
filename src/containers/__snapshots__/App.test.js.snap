// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app container 1`] = `ShallowWrapper {}`;
